### DISCUSSION TITLE
/nsm partition at 99% occupied disk space without using full available space

### DISCUSSION THREAD

**Discussion by bit1290** [Security-Onion-Solutions/securityonion#6456] [Unsupported Versions]
Hello,

We are getting our /nsm partition with 99% of used disk space as you can see from the following image:

![image](https://user-images.githubusercontent.com/16745902/144839766-36f266d1-7d51-4553-831d-0306ab52cf77.png)

But if we go to the directory /nsm and execute "du -shc ./*" command:

![image](https://user-images.githubusercontent.com/16745902/144839874-d98c1549-cacc-463b-ab7c-791fc33c0630.png)

We can see that the total disk space is 3.3T and we are only using 1.3T.

This doesn't make any sense.

This is our elastic configuration file:

[root@papidss003 suricata]#  cat /opt/so/saltstack/local/pillar/minions/papidss003_standalone.sls
elasticsearch:
  es_port: 9200
  esclustername: '{{ grains.host }}'
  esheap: 25000m
  log_size_limit: 1000
  mainint: enp2s0f0
  mainip: 10.135.101.138
  node_route_type: hot
  node_type: ''
host:
  mainint: enp2s0f0
idstools:
  config:
    oinkcode: ''
    ruleset: ETOPEN
    urls: null
  sids:
    disabled:
    - '2221010'
    - '2100368'
    - '2200074'
    - '2221045'
    - '2101411'
    - '2100366'
    - '1100012'
    enabled: null
    modify: null
kratos:
  kratoskey: du8SHNYmhSzBpQzR0OXw
logstash_settings:
  ls_input_threads: 1
  ls_pipeline_batch_size: 125
  ls_pipeline_workers: 80
  lsheap: 1000m
manager:
  domainstats: 0
  elastalert: 1
  es_port: 9200
  esclustername: '{{ grains.host }}'
  esheap: 25000m
  freq: 0
  grafana: 1
  mainint: enp2s0f0
  mainip: 10.135.101.138
  mtu: 1500
  no_proxy: localhost, 127.0.0.1, 10.135.101.138, papidss003
  osquery: 0
  playbook: 0
  proxy: http://10.135.101.133:3128
  thehive: 0
ntp:
  servers:
  - 10.220.3.113
  - 10.220.3.114
patch:
  os:
    enabled: true
    schedule_name: manual
    source: direct
    splay: 300
sensor:
  hnsensor: null
  interface: bond0
  mainint: enp2s0f0
  mainip: 10.135.101.138
  manager: ''
  mtu: 1500
  suriprocs: 19
  uniqueid: 1638456971
  zeek_lbprocs: null
sensoroni:
  node_address: 10.135.101.138
  node_description: SO
soc:
  es_index_patterns: '*:so-*,*:endgame-*'
steno:
  enabled: false

Could you help us with this situation?

Thank you.

**Comment by abesinger** [comment]
`du -s ./*` will not count any ["hidden" (a.k.a. "dot") files](https://en.wikipedia.org/wiki/Hidden_file_and_hidden_directory#Unix_and_Unix-like_environments).

Instead try `find . -maxdepth 1 | xargs du -ch` 

Now it's possible that space on `/nsm` is being occupied by a file that was unlinked but a process still has the file open. That could have happened if you `rm`ed an in-use file in attempt to recover disk space.  If you'd like more of an explanation of that situation, read [here](https://man7.org/linux/man-pages/man2/unlink.2.html).

There's no way to `ls` those unliked files (they don't have filenames)!

You can find them using `lsof` (you may have to install it. It's a nifty tool, I install it everywhere).
Try `lsof /nsm | fgrep "(deleted)"`

That will show you any process that have open filehandles on unlinked files under `/nsm`. If you find one of those, and want to recover the space occupied by it, you have to get the associated process close the filehandle, or you have to kill the process. Some modern Linux filesystems have the ability to recover recently deleted files (I'll leave how to do that as an exercise for the reader), but that won't recover space, it will just make the file visible again. However, I can imagine some scenarios where it would be better to let the process gracefully finish and then `rm` the file.

**Comment by jeffrylangvt** [comment]
We have the same issue and think we've pinpointed it to workers crashing and keeping the open file handles.  Sadly the only thing we've been able to do is to schedule a reboot of the sensors daily at our lowest volume time.

**Reply by abesinger** [reply]
So by "crashed," do you mean the worker died and exited, or do you mean something else?  Because if the process that has the filehandle exits, all of its filehandles get closed (technically, the filehandles no longer exist).

**Comment by Rasoul-Naghipour** [comment]
Hello
I'm using  SecurityOnion version 2.3 and I have the same problem in my forwarder nodes I fix this issue by using "sudo so-filebeat-restart" command on the server
or you can use the below commands at the Master node:
1. sudo salt-key  //to find the node id
2. sudo salt yournodeid cmd/run 'sudo so-filebeat-restart' //Please replace your node id with yournodeid


I am trying more to fix this problem from the ground up

**Reply by dougburks** [reply]
@Rasoul-Naghipour 

You might also want to review this older discussion:
https://github.com/Security-Onion-Solutions/securityonion/discussions/3852

But ultimately you should start a new discussion rather than replying to somebody else's discussion. From https://github.com/Security-Onion-Solutions/securityonion/discussions/1720:
> Start a new discussion instead of replying to somebody else's discussion
Please search to see if you can find similar discussions that may help you. However, in order to avoid confusion, please do NOT reply to somebody else's discussion with your own issue. Instead, please start a new discussion and in that new discussion you can provide a hyperlink to the related discussion.
